ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/role/RoleAuditFilter.java

```diff
@@ -76,7 +79,14 @@ public void setup()
     public boolean isFiltered(AuditEntry logEntry)
     {
         RoleResource primaryRole = RoleResource.role(logEntry.getUser());
-        return isFiltered(Roles.getRoles(primaryRole), logEntry.getPermissions(), logEntry.getResource());
+        try
+        {
+            return isFiltered(Roles.getRoles(primaryRole), logEntry.getPermissions(), logEntry.getResource());
+        }
+        catch (UncheckedExecutionException e)
+        {
+            throw Exceptions.tryGetCassandraExceptionCause(e);
```

Comment:
I would like to see a simple test case on this. Something like a CassandraException wrapped in a UncheckedExecutionException being thrown and the exception thrown outwards would be the CassandraException.

If it's hard to do (because it might be thrown by Roles.getRole(...)) then you can ignore it. But I think it could be nice to verify the behavior. I think this could be done by providing a "Function<RoleResource, Set<RoleResource>>" to the test constructor and then passing in "Roles::getRoles" from the default constructor.

I also noticed that isFiltered(AuditEntry) is not really unit tested by itself in TestRoleAuditFilter.